packages/marshal/src/helpers/environment-options.js

```diff
@@ -0,0 +1,131 @@
+ * with explicit parameters, and for the parameterization to be like
+ * calling the static module with parameters in order to derive from it a
+ * module instance. Compartments instead lets us parameterize the meaning
+ * of a module instance derived from a static module according to the
+ * three namespaces provided by the JavaScript semantics, effecting the
+ * meaning of a module instance.
+ *    * The global variable namespaces.
+ *       * The global scope, aliased to properties of the global object.
+ *         This is necessarily compartment-wide, and therefore in our
+ *         recommened usage pattern, packages-wide.
+ *       * The global lexical scope. The SES-shim compartments supports
+ *         these both compartment-wide as well as per-module. But it is
+ *         not yet clear what we will propose in the Compartment proposal.
+ *    * The import namespace.
+ *    * The host hooks.
+ *
+ * This `environment-options.js` module looks for a setting of of an
+ * `optionName` parameter rooted in the global scope. If follows the Node
+ * precedent for finding Unix environment variable settings, looking for a
+ * global `process` object holding an `env` object,
+ * optionally holding a property named for the `optionName` whose value is the
+ * configuration setting of that option. For example, for the optionName
+ * `ALLOW_IMPLICIT_REMOTABLES` it would look in
+ * `globalThis.process.env.ALLOW_IMPLICIT_REMOTABLES`.
+ *
+ * If setting is either absent or `undefined`, that indicates that
+ * this configuration option should have its default behavior, whatever that is.
+ * Otherwise, reflecting Unix environment variables, the setting must be a
+ * string. This also helps ensure that this channel is used only to pass data,
+ * not authority beyond the ability to read this global state.
+ *
+ * The current placement of this `environment-options.js` module in the
+ * `@agoric/marshal` package is a stopgap measure.
+ * TODO the intention is to migrate it into Endo, and to migrate all our
+ * direct uses of `process.env` for configuration parameters to use it
+ * instead.
+ *
+ * Even after that migration, this module will still not be used for
+ * `LOCKDOWN_OPTIONS` itself, since that must happen before `lockdown`,
+ * whereas this module must initialize after `lockdown`.
+ *
+ * @param {string} optionName
+ * @param {string=} defaultSetting
+ */
+export const getEnvironmentOption = (
+  optionName,
+  defaultSetting = undefined,
+) => {
+  assert.typeof(
+    optionName,
+    'string',
+    X`Environment option name ${q(optionName)} must be a string.`,
+  );
+  assert(
+    defaultSetting === undefined || typeof defaultSetting === 'string',
+    X`Environment option default setting ${q(
+      defaultSetting,
+    )}, if present, must be a string.`,
+  );
+
+  let setting = defaultSetting;
+  const globalProcess = globalThis.process;
+  if (typeof process === 'object' && typeof globalProcess.env === 'object') {
```

Comment:
If `process` is exactly `null`, the second check will produce a TypeError.

Reply:
Done. Resolving.